Weapon Transmog

BarleyFlour opened the pull request:
I've added the ability to transmog your weapons. (Its own "tab" at the bottom of the menu, see attached image)
Currently by slot and set, as well as being limited to weapons of the same WeaponAnimationStyle.
The model-switching is achieved by patching the getter for UnitViewHandSlotData.VisibleItemBlueprint.
This does however mean that sound and visual effects of attacks don't get matched.

Please feel free to tell me about any viewpoints/suggestions, as I'm somewhat bad at IMGUI I'm sure it could do with some improvement. (On here or Discord, whichever you prefer) 
![WeaponTransmog](https://github.com/xADDBx/RTTransmog/assets/82173938/764ea90d-7130-4b92-ac19-11b69c7810d1)

BarleyFlour committed:
Initial Push, Added rudimentary weapon transmogging capability, not yet assignable by weapon set or slot.

BarleyFlour committed:
Removed empty weapon anim style categories.
Fixed "show all" sorting for weapons, now only shows weapons of appropriate anim style, instead of all weapons.

BarleyFlour committed:
Added Main/Offhand selection, aswell as weapon set selection.

BarleyFlour committed:
Minor Weapon Set selector ui stuff.

BarleyFlour committed:
Moved hand selection into WeaponSlotSelector

BarleyFlour committed:
Remove commented initialization.

BarleyFlour committed:
Removed Unnecessary imports in Main. aswell as commented out things.

BarleyFlour committed:
Merge remote-tracking branch 'origin/master'

BarleyFlour committed:
Changed braces to align with original repo.

BarleyFlour committed:
More formatting.

BarleyFlour committed:
Even more formatting.

BarleyFlour committed:
Too much formatting.

BarleyFlour committed:
Infinite formatting.

BarleyFlour committed:
Fixed "Show All" for weapons.

xADDBx commented:
So
- Hiding irrelevant Weapon Browsers would help with UI
- As mentioned, Patches.cs#55 can cause an NRE for Mainhand/Offhand Slots.
- For whatever reason this neither persists through saving nor actually transmogs those weapons for fights? When I create the override, it'll show that weapons changed in the doll room. But during actual fights the characters will still take the original weapon?

xADDBx commented:
Screenshots of it not working during actual combat (despite working in doll room):
![grafik](https://github.com/xADDBx/RTTransmog/assets/62178123/34267580-0528-47ac-b5cb-fd1c6de82f9a)
![grafik](https://github.com/xADDBx/RTTransmog/assets/62178123/99bde204-0d65-4b88-9571-c95b9950ffec)
![grafik](https://github.com/xADDBx/RTTransmog/assets/62178123/68521363-8d7f-472d-b2f7-4ec4d1de74a8)
Screenshot of option not persisting through quicksave+load:
![grafik](https://github.com/xADDBx/RTTransmog/assets/62178123/39241dde-7e46-4da1-b10b-f78d80362fba)

Log:
[GameLogFull.txt](https://github.com/user-attachments/files/15806433/GameLogFull.txt)
Other Active Mods:
ToyBox, MicroPatches, UnityExplorer
Save: 
[UnpackMe.zip](https://github.com/user-attachments/files/15806498/UnpackMe.zip)

BarleyFlour committed:
Made only relevant browser appear unless a toggle is checked to show all.

BarleyFlour committed:
Fixed a NullReferenceException in
Patches.
Fixed m_Weapons not being saved.
Fixed animation relevant browser "sticking".

BarleyFlour committed:
Fixed weapons on your character model not updating immediately.

BarleyFlour committed:
Fixed weapons not updating immediately, again.

xADDBx committed:
No Magic Numbers

xADDBx committed:
Workshop description update

xADDBx committed:
Version Update

xADDBx committed:
Put logging behind #if DEBUG flags

xADDBx commented:
Looks good to me; I did some adjustments (mostly version changes) in https://github.com/BarleyFlour/RTTransmog/pull/1
After you merged those, I'll merge this and create a release if you don't want to do further changes.

BarleyFlour committed:
Merge pull request #1 from xADDBx/MinorAdjustments

Minor adjustments
